### PR TITLE
[HYDRA-343] [HYDRA-369] [CDAP-7025] fix more than 2 input stages for Join

### PR DESCRIPTION
**Pull request by elfenheart**
- Handle case when there are more than 2 input stages into Joiner 
- Add bottom padding to rule box
- Remove conditions for disconnected stage

JIRAs:
https://issues.cask.co/browse/HYDRA-343
https://issues.cask.co/browse/HYDRA-369
https://issues.cask.co/browse/CDAP-7025

Build: http://builds.cask.co/browse/CDAP-DRC4234-1

### CODE REVIEW

File: cdap-ui/app/directives/widget-container/widget-sql-conditions/widget-sql-conditions.js [hunk 1]
```diff
@@ -82,6 +79,10 @@ function SqlConditionsController() {
         return field.name;
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/widget-container/widget-sql-conditions/widget-sql-conditions.js]: There seems to be `JSON.parse` in `initializeOptions` . We should probably wrap that in a `try...catch` block

**Comment by ajainarayanan** [comment]
#### TL;DR

  Adding new stages along with existing stages to joiner plugin doesn't by default choose all the fields.

#### Longer version

When we add a new stage its by default not selected (after adding initial say 2 stages). The confusing part is the stage is already present in the conditions with an empty dropdown (meaning no field is selected from that stage). 
- Add two streams and connect them to the joiner
- Open joiner and all fields are by default selected and it shows proper error and the join condition already has both the stages with first field selected
- Resolve the error and close joiner
- Add two KVTables and open the joiner. 
1. No error shown
2. Sql condition has the stage name with empty dropdown
3. I select all the fields one SnapshotAvro and choose the field from dropdown in join condition and close the joiner
4. Re-open the joiner and now both the SnapshotAvro dropdowns are filled in the join condition but the second SnapshotAvro still has no fields selected.

Not sure if its fixable entirely but we should atleast make the stages have all fields selected by default whenever added.

**Comment by elfenheart** [comment]
@ajainarayanan the empty dropdown is by design. Need to indicate to user that something has changed. If we fill in by default, when they have changed the connection, it's not immediately obvious. We could add an error message I suppose

**Comment by ajainarayanan** [comment]
👍 LGTM
